src/plugins/markdown-with-image/markdown-with-image-display.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import classNames from 'classnames';
+import { IMAGE_POSITION } from './constants.js';
+import Markdown from '../../components/markdown.js';
+import ClientConfig from '../../bootstrap/client-config.js';
+import { getAccessibleUrl } from '../../utils/source-utils.js';
+import { useService } from '../../components/container-context.js';
+import CopyrightNotice from '../../components/copyright-notice.js';
+import { sectionDisplayProps } from '../../ui/default-prop-types.js';
+
+export default function MarkdownWithImageDisplay({ content }) {
+  const clientConfig = useService(ClientConfig);
+  const { text, image } = content;
+
+  const imageSrc = getAccessibleUrl({ url: image.sourceUrl, cdnRootUrl: clientConfig.cdnRootUrl });
+  const imageWrapperStyle = { float: image.position };
+  if (image.position === IMAGE_POSITION.left) {
+    imageWrapperStyle.paddingLeft = 0;
+  } else {
+    imageWrapperStyle.paddingRight = 0;
+  }
+
+  return (
+    <div className="MarkdownWithImageDisplay">
+      <div className="MarkdownWithImageDisplay-verticallyStretchedContent">
```

Comment:
I haven't worked with `float`ing HTML since early childhood, so I was not sure if the "clearfix hack" one needed back then was still a thing. Obviously it's not, but there is something more modern: https://stackoverflow.com/a/31954550. And it seems to be supported by all browsers: https://caniuse.com/?search=flow-root

We could use that just to follow current practices, I guess, in reality it probably won't make any visual difference, I guess.

Reply:
Nice, works, thanks